wordpress-mobile/WordPress-iOS
# Offline Mode: Switch to local feature flag

## Description
* Switches `syncPublishing` flag to a local feature flag
* Adds `FeatureFlagRolloutStore` to facilitate rolling out the flag to a specified percentage of users

## How to test

- Change `FeatureFlag.swift` L45 to `return FeatureFlagRolloutStore().isRolloutEnabled(for: self)` in order to test the percentage rollout mechanism
- Set a breakpoint on L43 of `FeatureFlagRolloutStore.swift`
- Run the app with the `syncPublishing` rollout percentage set to 1% 
- ✅ The sync publishing feature flag should disabled (highly likely, but not impossible that it's enabled if you make it into the 1% rollout group) 
- When the app hits the breakpoint, note the assigned rollout group and release the breakpoint
- Run the app with the `syncPublishing` rollout percentage set to a value higher than `assigned rollout group / 10` (e.g. if the assigned group was 120, set the rollout percentage to 13)
- ✅ The sync publishing feature flag should enabled  

## Regression Notes
1. Potential unintended areas of impact
sync publishing feature flag

2. What I did to test those areas of impact (or what existing automated tests I relied on)
tested manually

3. What automated tests I added (or what prevented me from doing so)
none

PR submission checklist:
